pyface/tasks/task_application.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2014-2016 by Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in enthought/LICENSE.txt and may be redistributed only
+# under the conditions described in the aforementioned license.  The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+# Thanks for using Enthought open source!
+
+""" Define a base Task application class to create the event loop, and launch
+the creation of tasks and corresponding windows.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import logging
+import platform
+
+from traits.api import (
+    List, Instance, Property, cached_property, on_trait_change
+)
+
+from pyface.application import Application
+from .task_window import TaskWindow
+
+IS_WINDOWS = platform.system() == 'Windows'
+logger = logging.getLogger(__name__)
+
+
+class TaskApplication(Application):
+    """ A base class for Pyface tasks applications.
+
+    This handles setting up logging, starting up the GUI, and other common
+    features that we want when creating a GUI application.
+    """
+
+    # -------------------------------------------------------------------------
+    # 'TaskGuiApplication' interface
+    # -------------------------------------------------------------------------
+
+    # Tasks management --------------------------------------------------------
+
+    #: List of all running tasks
+    tasks = List("pyface.tasks.task.Task")
```

Comment:
New bug found: this will lead toy a list of unicode elements, instead of a list of Task instances! Missing `Instance`.

Reply:
Yep, found that this morning...